### DISCUSSION TITLE
How to properly close the client connection on the application exiting?

### DISCUSSION THREAD

**Discussion by AnderBarDreams** [jchristn/SuperSimpleTcp#67] [Q&A]
Given a client that is connected to the server like this:

```
var client = new SimpleTcpClient(ipAddress, port);
client.Connect();
```

I would like to close the tcp connection when the application is shutting down. Should I call both `client.Disconnect()` and `client.Dispose()` or just `client.Disconnect()` is enough?

**Comment by jchristn** [comment]
Hi @AnderBarDreams, ```.Disconnect()``` will not call ```.Dispose()``` intentionally - that way you have the freedom to disconnect and connect as needed.

The best approach is to either call ```.Dispose()``` when done, or better, use a ```using``` block, i.e.
```
using (SimpleTcpClient client = new SimpleTcpClient(ip, port))
{
  // do stuff
}
```
Cheers, Joel
